rest/src/main/java/org/n52/io/extension/RenderingHintsExtension.java

```diff
@@ -74,7 +74,11 @@ public void addExtraMetadataFieldNames(DatasetOutput output) {
     }
 
     private boolean hasRenderingHints(DatasetOutput output) {
-        return hasSeriesConfiguration(output) || hasPhenomenonConfiguration(output);
+        return hasDatasetParameters(output) && (hasSeriesConfiguration(output) || hasPhenomenonConfiguration(output));
+    }
+
+    private boolean hasDatasetParameters(DatasetOutput output) {
+        return output.getDatasetParameters() != null;
```

Comment:
this might have a side effect. think of the query `?fields=extras` which wouldn't have `renderingHints` included (as `DatasetOutput#datasetParameters` are `null`).

We have to think of how extensions will work even when `fields` parameter filters out some output needed by the extension:
1. either rework the way extensions (using fieldable `AbstractOutput` data) gets information needed
2. create some kind of `AbstractOutput` wrapping containing a flag if data shall be serialized or not (extensions could then force getting the data even if it is not flagged to be serialized later)

This targets for example `StatusIntervalExtension` as well.

what do you think?